core/src/main/java/org/codice/imaging/nitf/core/image/impl/RasterProductFormatUtilities.java

```diff
@@ -59,25 +66,42 @@ public RasterProductFormatUtilities() throws NitfFormatException {
     }
 
     /**
-        Unmarshal input stream to RPFs list.
-
-        @param inputStream the input stream to parse.
-        @throws JAXBException if the unmarshalling of the XML fails.
-    */
+     * Unmarshal input stream to RPFs list.
+     *
+     * @param inputStream the input stream to parse.
+     * @throws JAXBException if the unmarshalling of the XML fails.
+     */
     private void unmarshal(final InputStream inputStream) throws JAXBException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+        Document document = null;
+        try {
+            dbf.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            dbf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+            dbf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+            dbf.setExpandEntityReferences(false);
+            DocumentBuilder db = dbf.newDocumentBuilder();
+            document = db.parse(inputStream);
+        } catch (ParserConfigurationException e) {
+            LOG.debug("Could not set features on {}", dbf);
```

Comment:
I think i probably want to break this out. The `ParserConfigurationException` is thrown by each of the `setFeature` calls, as well as the `newDocumentBuilder` call. I don't think the `setFeature` calls failing would cause any issues, however, not having a good `documentBuilder` would. I'll log and throw a new exception if the `newDocumentBuilder` fails.